gremlin-javascript/src/main/javascript/gremlin-javascript/lib/process/anonymous-traversal.js

```diff
@@ -31,13 +31,17 @@ const Graph = require('../structure/graph').Graph;
  */
 class AnonymousTraversalSource {
 
+  constructor(traversalSourceClass) {
+    this.traversalSourceClass = traversalSourceClass;
+  }
+
   /**
    * Constructs an {@code AnonymousTraversalSource} which will then be configured to spawn a
    * {@link GraphTraversalSource}.
    * @returns {AnonymousTraversalSource}.
    */
-  static traversal() {
-    return new AnonymousTraversalSource();
+  static traversal(traversalSourceClass) {
```

Comment:
We should document in jsdoc: `@param {GraphTraversalSource} [traversalSourceClass] Optional GraphTraversalSource constructor.`